File: cart-service/src/main/java/org/vtsukur/graphql/demo/cart/web/http/CartWithProductsProjection.java
package org.vtsukur.graphql.demo.cart.web.http;

import lombok.Data;

import java.math.BigDecimal;
import java.util.List;

@Data
public class CartWithProductsProjection {

    private Long id;

    private List<Item> items;

    private BigDecimal subTotal;

    @Data
    public static class Item {

        private ProductProjection product;

        private int quantity;

        private BigDecimal total;

        @Data
        public static class ProductProjection {

            private String id;

            private String title;

            private BigDecimal price;

            private String image;

            private String sku;

        }

    }

}

File: cart-service/src/main/kotlin/org/vtsukur/graphql/demo/cart/web/http/CartWithProductsProjection.kt
package org.vtsukur.graphql.demo.cart.web.http

import java.math.BigDecimal

data class CartWithProductsProjection(var id: Long = -1L, var items: List<Item> = mutableListOf(),
                                      var subTotal: BigDecimal = BigDecimal.ZERO) {

    data class Item(var product: ProductProjection = ProductProjection(), var quantity: Int = 0,
                    var total: BigDecimal = BigDecimal.ZERO) {

        data class ProductProjection(var id: String = "", var title: String = "", var price: BigDecimal = BigDecimal.ZERO,
                                     var image: String = "", var sku: String = "")
    }
}


